drivers/can/can_nxp_s32_canxl.c

```diff
@@ -0,0 +1,977 @@
+		can_nxp_s32_ctrl_callback(dev, eventType, buffIdx, canexcelState);		\
+	}
+
+#if defined(CONFIG_CAN_FD_MODE)
+#define CAN_NXP_S32_TIMING_DATA_CONFIG(n)							\
+		.bitrate_data = DT_PROP(CAN_NXP_S32_NODE(n), bus_speed_data),			\
+		.sjw_data = DT_PROP(CAN_NXP_S32_NODE(n), sjw_data),				\
+		.prop_seg_data = DT_PROP_OR(CAN_NXP_S32_NODE(n), prop_seg_data, 0),		\
+		.phase_seg1_data = DT_PROP_OR(CAN_NXP_S32_NODE(n), phase_seg1_data, 0),		\
+		.phase_seg2_data = DT_PROP_OR(CAN_NXP_S32_NODE(n), phase_seg2_data, 0),		\
+		.sample_point_data = DT_PROP_OR(CAN_NXP_S32_NODE(n), sample_point_data, 0),
+#define CAN_NXP_S32_FD_MODE	1
+#define CAN_NXP_S32_BRS		CONFIG_CAN_NXP_S32_BRS
+#else
+#define CAN_NXP_S32_TIMING_DATA_CONFIG(n)
+#define CAN_NXP_S32_FD_MODE	0
+#define CAN_NXP_S32_BRS		0
+#endif
+
+#define CAN_NXP_S32_INIT_DEVICE(n)								\
+	CAN_NXP_S32_CTRL_CALLBACK(n)								\
+	CAN_NXP_S32_ERR_CALLBACK(n)								\
+	CAN_NXP_S32_IRQ_CONFIG(n)								\
+	PINCTRL_DT_DEFINE(CAN_NXP_S32_NODE(n));							\
+	Canexcel_Ip_ConfigType can_nxp_s32_default_config##n = {				\
+		.rx_mbdesc = (uint8)CAN_NXP_S32_MAX_RX,						\
+		.tx_mbdesc = (uint8)CAN_NXP_S32_MAX_TX,						\
+		.CanxlMode = CANEXCEL_LOOPBACK_MODE,						\
+		.fd_enable = (boolean)CAN_NXP_S32_FD_MODE,					\
+		.bitRateSwitch = (boolean)CAN_NXP_S32_BRS,					\
+		.ctrlOptions = (uint32)CANXL_IP_BUSOFF_RECOVERY_U32,				\
+		.Callback = nxp_s32_can_##n##_ctrl_callback,					\
+		.ErrorCallback = nxp_s32_can_##n##_err_callback					\
+	};											\
+	__nocache Canexcel_Ip_StateType can_nxp_s32_state##n;					\
+	__nocache Canexcel_TxFdMsgType tx_msg##n[CAN_NXP_S32_MAX_TX];				\
+	__nocache Canexcel_RxFdMsg rx_msg_##n[CAN_NXP_S32_MAX_RX];				\
+	static struct can_nxp_s32_data can_nxp_s32_data_##n = {					\
+		.can_state = (Canexcel_Ip_StateType *)&can_nxp_s32_state##n,			\
+		.tx_msg = tx_msg##n,								\
+		.rx_msg = rx_msg_##n,								\
+	};											\
+	static struct can_nxp_s32_config can_nxp_s32_config_##n = {				\
+		.base_sic = (CANXL_SIC_Type *)DT_REG_ADDR_BY_NAME(CAN_NXP_S32_NODE(n), sic),	\
+		.base_grp_ctrl = (CANXL_GRP_CONTROL_Type *)					\
+					DT_REG_ADDR_BY_NAME(CAN_NXP_S32_NODE(n), grp_ctrl),	\
+		.base_dsc_ctrl = (CANXL_DSC_CONTROL_Type *)					\
+					DT_REG_ADDR_BY_NAME(CAN_NXP_S32_NODE(n), dsc_ctrl),	\
+		.instance = n,									\
+		.clock_can = DT_PROP(CAN_NXP_S32_NODE(n), clock_frequency),			\
+		.bitrate = DT_PROP(CAN_NXP_S32_NODE(n), bus_speed),				\
+		.sjw = DT_PROP(CAN_NXP_S32_NODE(n), sjw),					\
+		.prop_seg = DT_PROP_OR(CAN_NXP_S32_NODE(n), prop_seg, 0),			\
+		.phase_seg1 = DT_PROP_OR(CAN_NXP_S32_NODE(n), phase_seg1, 0),			\
+		.phase_seg2 = DT_PROP_OR(CAN_NXP_S32_NODE(n), phase_seg2, 0),			\
+		.sample_point = DT_PROP_OR(CAN_NXP_S32_NODE(n), sample_point, 0),		\
+		CAN_NXP_S32_TIMING_DATA_CONFIG(n)						\
+		.max_bitrate = DT_CAN_TRANSCEIVER_MAX_BITRATE(CAN_NXP_S32_NODE(n),		\
+								CAN_NXP_S32_MAX_BITRATE),	\
+		.phy = NULL,									\
```

Comment:
Shouldn't this be a reference to the PHY?

Reply:
yes, I updated